### PR TITLE
[Vue] migrate piwik-widget and related directives

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

**This PR is based off of #18428.**

Changes:
- Migrate piwik-widget-bydimension-container directive.
- Migrate piwik-widget-container directive.
- Migrate piwik-widget directive.
- Do not show ajax request failures if not logged in.
- Remove SiteSelector adapter hack since it is fixed by the Field adapter ng model binding + remove selectedSite state since it's not used anymore.
- Create v-tooltips directive to encapsulate jquery UI tooltips usage. Useful since we want to remove it when an element disappears, not the component.
- Add Widgets store for handling results of widgetsHelper.getAvailableWidgets().

image

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by diosmosis** [comment]
UI tests should be passing for this PR

File: plugins/CoreHome/stylesheets/layout.less [hunk 1]
```diff
@@ -190,7 +190,7 @@ nav {
       padding-bottom: 20px;
 
       .menu-icon {
-        padding-right: 13px;
+        padding-right: 9.5px;
```
Comment (sgiehl) [inline plugins/CoreHome/stylesheets/layout.less]: I guess this one, will be there now in every branch, that's not freshly branched from `4.x-dev`
```suggestion
        padding-right: 13px;
```

File: plugins/Dashboard/tests/UI/DashboardManager_spec.js [hunk 2]
```diff
@@ -98,7 +98,7 @@ describe("DashboardManager", function () {
 
         expect(await page.screenshot({ fullPage: true })).to.matchImage('create_new');
     });
-
+return;
```
Comment (sgiehl) [inline plugins/Dashboard/tests/UI/DashboardManager_spec.js]: same here...

**Comment by diosmosis** [comment]
@sgiehl I'll fix these in the other branches individually. I will have limited availability however for the next couple weeks, so it's ok to do one Vue review a day (if there's a PR to review) in my opinion. I won't have time to get to everything.

**Review by sgiehl — APPROVED** [review]
Code looks fine and I wasn't able to find any related issues while clicking through the UI

**Pull request merged**